docs/misc/deprecated.md

```diff
@@ -22,6 +22,9 @@ variants:
 
     docker run -c (--cpu-shares)
 
+`docker run --volume-driver` is now deprecated in favor of `docker volume
```

Comment:
Early docs review; This needs its own head, right now it's below "command line short variant options"